fpga_interchange/fasm_generators/xc7.py

```diff
@@ -197,21 +294,69 @@ def handle_slice_routing_bels(self):
             slice_prefix = self.get_slice_prefix(site, tile_type,
                                                  sites_in_tile)
 
-            routing_bel_feature = "{}.{}.{}.{}".format(tile_name, slice_prefix,
-                                                       bel, pin)
-            self.cells_features.append(routing_bel_feature)
+            if bel in used_muxes:
+                if pin not in ["0", "1"]:
+                    self.add_cell_feature((tile_name, slice_prefix, bel))
+            else:
+                self.add_cell_feature((tile_name, slice_prefix, bel, pin))
+
+    def handle_pips(self):
+
+        PipRegex = namedtuple('PipRegex', 'regex callback')
+
+        regexs = list()
+        regexs.append(
+            PipRegex(
+                regex="(CLK_HROW_CK_IN_[LR][0-9]+)",
+                callback=lambda m: ["{}_ACTIVE".format(m.group(1))]))
+        regexs.append(
+            PipRegex(
+                regex="(CLK_HROW_R_CK_GCLK[0-9]+)",
+                callback=lambda m: ["{}_ACTIVE".format(m.group(1))]))
+        regexs.append(PipRegex(
+            regex="(HCLK_CMT_CCIO[0-9]+)",
+            callback=lambda m: ["{}_{}".format(m.group(1), f) for f in ["ACTIVE", "USED"]]
+            ))
+        regexs.append(PipRegex(
+            regex="CLK_BUFG_REBUF_R_CK_(GCLK[0-9]+)_BOT",
+            callback=lambda m: ["{}_{}".format(m.group(1), f) for f in ["ENABLE_BELOW", "ENABLE_ABOVE"]]
+            ))
+
+        tile_types = [
+            "HCLK_L", "HCLK_R", "HCLK_L_BOT_UTURN", "HCLK_R_BOT_UTURN",
+            "HCLK_CMT", "HCLK_CMT_L", "CLK_HROW_TOP_R", "CLK_HROW_BOT_R",
+            "CLK_BUFG_REBUF"
+        ]
+        extra_features = dict((tile_type, list()) for tile_type in tile_types)
+
+        site_thru_pips = self.fill_pip_features(extra_features)
+
+        import sys
```

Comment:
Doing an "import sys" seems wrong?  Not even sure what it is doing here?

Reply:
Leftover, removed